[shopsys] removed previously deprecated features

grossmannmartin opened the pull request:
| Q             | A
| ------------- | ---
|Description, reason for the PR| all previously deprecated features are removed in this PR
|New feature| No <!-- Do not forget to update docs/ -->
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes <!-- Do not forget to update UPGRADE.md -->
|Fixes issues| ... <!-- Write "closes #123" for the issue to be closed automatically during merge -->
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

RobinDvorak commented on packages/framework/src/Component/FileUpload/FileUpload.php:
```diff
@@ -51,7 +48,7 @@ class FileUpload
-     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface|null
+     * @var \Symfony\Component\DependencyInjection\ParameterBag\ParameterBagInterface
      */
     protected $parameterBag;
```
I think its safe to set explicit type here and for all other properties. But it just nice to have in this PR.
```suggestion
    protected ParameterBagInterface $parameterBag;
```

grossmannmartin replied:
please let's move typed properties into a separate PR

RobinDvorak reviewed (approved):
I am approving this with one comment and its up to you, if you want to change it or resolve it without changes. I dont think its necessary to change it in this PR. What I cant review properly are upgrade notes :D It looks ok to me, but...

vitek-rostislav commented on upgrade/UPGRADE-v11.0.0-dev.md:
````diff
@@ -220,3 +220,4 @@ There you can find links to upgrade notes for other versions too.
                 )
             ```
     - method `Shopsys\ReadModelBundle\Image\ImageViewFacade::getForEntityIds()` was removed, use `getMainImagesByEntityIds()` instead
+    - method `Shopsys\ReadModelBundle\Product\Action\ProductActionViewFacade::getForArray()` was removed, use `ProductActionViewFactory` instead
````
I would `::createFromArray()` here :slightly_smiling_face:

grossmannmartin replied:
added in 89bb0f956

vitek-rostislav commented on packages/framework/src/Component/Router/FriendlyUrl/FriendlyUrlGenerator.php:
```diff
@@ -90,20 +79,7 @@ public function generateFromRouteCollection(
 
         $domainId = $domainConfig->getId();
 
-        if ($this->mainFriendlyUrlSlugCache !== null) {
```
I am confused here - why is this block removed, please? From now on, `mainFriendlyUrlSlugCache` should be always defined, so I would expect you to remove the `else` block, not the `if` block, or am I missing something? :thinking:

grossmannmartin replied:
No, you don't miss anything... This is my mistake. Thanks for pointing it out! fixed in 9bcc51f99

vitek-rostislav commented on upgrade/UPGRADE-v11.0.0-dev.md:
````diff
@@ -503,6 +503,17 @@ There you can find links to upgrade notes for other versions too.
                 )
             ```
+    - `Shopsys\FrontendApiBundle\Controller\FrontendApiController`
+        - property `$graphqlConfigurator` is no longer nullable
````
it seems like it has never been nullable (maybe this applies to many and many other properties mentioned here in the upgrade notes in the previous commits, but I think the notes are good enough now, I do not want you to go through them again :smile: )

grossmannmartin replied:
You're right. They were not annotated as nullable, but in some scenarios, they could be null (overridden constructor).

So from my PoV, it seems more like an incorrect annotation before.

vitek-rostislav commented on project-base/src/Controller/Front/ErrorController.php:
```diff
@@ -46,25 +46,33 @@ class ErrorController extends FrontBaseController
+    /**
+     * @var string|null
+     */
+    protected ?string $overwriteDomainUrl;
```
since this is in the `project-base` package, it should be `private` :innocent:

grossmannmartin replied:
👍 85ab620da

vitek-rostislav commented on project-base/scripts/configure.sh:
```diff
@@ -37,9 +37,7 @@ if [[ -d "project-base" ]]; then
-echo "Creating config files.."
-cp -f "${projectPathPrefix}config/parameters.yaml.dist" "${projectPathPrefix}config/parameters.yaml"
-cp -f "${projectPathPrefix}config/parameters_test.yaml.dist" "${projectPathPrefix}config/parameters_test.yaml"
+echo "Creating domain file.."
```
nitpick, but I would be rather more specific...
```diff
- echo "Creating domain file.."
+ echo "Creating domains_urls.yaml file.."
```

grossmannmartin replied:
823ee2eb3 it is

vitek-rostislav reviewed (changes requested):
Hi, this is one hell of a work, excellent job, you have my respect! :muscle: I have added a few notes, so please, check them out. Also, I would say I am missing upgrade notes for the last commit (`parameter(_test).yaml` files removal) - at least the "project-base-diff" should be there IMHO + mention about the possibility of creating the custom `parameters.yaml` file would be nice there :slightly_smiling_face:

grossmannmartin commented on packages/framework/src/Model/Breadcrumb/ErrorPageBreadcrumbGenerator.php:
```diff
@@ -45,24 +34,12 @@ public function getRouteNames()
-    {
-        DeprecationHelper::triggerAbstractMethod(__METHOD__);
-
-        return t('Page not found');
```
thanks! you're right. Translations removed in 2b8373350

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -66,3 +66,20 @@ There you can find links to upgrade notes for other versions too.
+        - `getMessageForNoLongerAvailableProduct()`
+        - `getMessageForChangedProduct()`
+    - class `Shopsys\FrameworkBundle\Model\LegalConditions\LegalConditionsFacade` was changed to abstract, don't forget to extend this class and implement following methods:
+        - `getTermsAndConditionsDownloadFilename();`
```
on Github, it's not possible when you're viewing a single commit... it bothers me too... 

fixed in 31c4346b8

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
````diff
@@ -104,3 +104,5 @@ There you can find links to upgrade notes for other versions too.
         ```
     - method `Shopsys\FrameworkBundle\Model\Pricing\PriceConverter::convertPriceWithoutVatToPriceInDomainDefaultCurrency()` was removed, use `convertPriceWithoutVatToDomainDefaultCurrencyPrice'()` instead
     - method `Shopsys\FrameworkBundle\Model\Pricing\PriceConverter::convertPriceWithVatToPriceInDomainDefaultCurrency()` was removed, use `convertPriceWithVatToDomainDefaultCurrencyPrice'()` instead
+    - method `Shopsys\FrameworkBundle\Model\Product\ProductFacade::getSellableByUuid` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product` namespace
````
good idea. Added in c11d2da6b and 97e29f775

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+        - method `getProductsOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+        - method `getProductsByCategory()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacade`
+        - method `getProductsCountOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
```
🤔 actually it isn't... this line is mentioned three times, but it's for 
- ProductOnCurrentDomainElasticFacade
- ProductOnCurrentDomainFacade
- ProductOnCurrentDomainFacadeInterface

but you pointed to me another issue... I recommend using a method that is removed a few lines later 😄 

changed in 4ed7faaa2

vitek-rostislav replied:
ah, ok :smile:

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+        - method `getProductsByCategory()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacade`
+        - method `getProductsCountOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+        - method `getProductsOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
```
same as above

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacade`
+        - method `getProductsCountOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+        - method `getProductsOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+        - method `getProductsByCategory()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
```
same as above above

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+        - method `getProductsOnCurrentDomain()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+        - method `getProductsByCategory()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface`:
+        - method `getProductsCountOnCurrentDomain()` was removed
```
same as above above above (this is interface only, so no replacement is suggested)

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+        - method `getProductsByCategory()` was removed, use method with the same name from the `Shopsys\FrontendApiBundle\Model\Product\ProductFacade` class
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface`:
+        - method `getProductsCountOnCurrentDomain()` was removed
+        - method `getProductsOnCurrentDomain()` was removed
```
same as above

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -108,3 +108,20 @@ There you can find links to upgrade notes for other versions too.
+    - `Shopsys\FrameworkBundle\Model\Product\ProductOnCurrentDomainFacadeInterface`:
+        - method `getProductsCountOnCurrentDomain()` was removed
+        - method `getProductsOnCurrentDomain()` was removed
+        - method `getProductsByCategory()` was removed
```
same as above

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -134,3 +134,4 @@ There you can find links to upgrade notes for other versions too.
         - method `getProductsByCategoryCount()` was removed, use `getFilteredProductsByCategoryCount()` instead
         - method `getProductsByBrand()` was removed, use `getFilteredProductsByBrand()` instead
         - method `getProductsByBrandCount()` was removed, use `getFilteredProductsByBrandCount()` instead
+    - removed parameter `shopsys.var_dir` from shopsys/framework package, add this parameter to your project config
```
🆒  added in 4ea0e3346

grossmannmartin commented on upgrade/UPGRADE-v11.0.0-dev.md:
```diff
@@ -128,6 +128,7 @@ There you can find links to upgrade notes for other versions too.
     - constant `Shopsys\FrameworkBundle\Model\Product\BestsellingProduct\BestsellingProductFacade::MAX_RESULTS` was removed
     - method `Shopsys\FrameworkBundle\Model\Script\ScriptFacade::getAllPagesScriptCodes()` was removed, use `getAllPagesBeforeContentScriptCodes()` instead
     - all unnecessary service definitions were removed from `services_test.yaml` file
+    - removed `Shopsys\FrameworkBundle\Component\FileUpload\DeleteOldUploadedFilesCronModule` cron definition from shopsys/framework package, add this cron to your project config
```
sir, yes, sir :) 82177e631

grossmannmartin commented:
> Hi, this is one hell of a work, excellent job, you have my respect! 💪 I have added a few notes, so please, check them out. Also, I would say I am missing upgrade notes for the last commit (`parameter(_test).yaml` files removal) - at least the "project-base-diff" should be there IMHO + mention about the possibility of creating the custom `parameters.yaml` file would be nice there 🙂

Added in b1b73941c, hope it's understandable
